### PR TITLE
Split ServerRequest::env() into getter/setter

### PR DESCRIPTION
**Pull request by cleptric**
Continues #9978

### CODE REVIEW

File: src/Http/ServerRequest.php [hunk 1]
```diff
@@ -1743,9 +1743,44 @@ public function withProtocolVersion($version)
...
+    }
+
+    /**
+     * Set a value to the request's environment data.
+     *
+     * @param string $key The key you want to write to.
+     * @param string $value Value to set
+     * @return $this
+     */
+    public function setEnv($key, $value)
```
Comment (markstory) [inline src/Http/ServerRequest.php]: Given that requests are trending towards immutable patterns, shouldn't any new methods follow that style?

File: tests/TestCase/Http/ServerRequestTest.php [hunk 2]
```diff
@@ -3520,20 +3520,30 @@ public function testWithRequestTarget()
...
+     * @return void
+     */
+    public function testGetEnv()
+    {
+        $request = new ServerRequest();
 
         //Test default null
-        $request->setEnv('HTTP_HOST');
-        $this->assertSame($request->getEnv('HTTP_HOST'), null);
+        $this->assertSame($request->getEnv('Foo'), null);
```
Comment (markstory) [inline tests/TestCase/Http/ServerRequestTest.php]: You should use assertNull().

File: tests/TestCase/Http/ServerRequestTest.php [hunk 3]
```diff
@@ -3543,7 +3543,7 @@ public function testGetEnv()
         $request = new ServerRequest();
 
         //Test default null
-        $this->assertSame($request->getEnv('Foo'), null);
+        $this->assertNull($request->getEnv('Foo'), null);
```
Comment (dereuromark) [inline tests/TestCase/Http/ServerRequestTest.php]: The 2nd argument should then be removed.

File: tests/TestCase/Http/ServerRequestTest.php [hunk 4]
```diff
@@ -3543,7 +3543,7 @@ public function testGetEnv()
...
 
         //Test default set
         $this->assertSame($request->getEnv('Foo', 'Bar'), 'Bar');
```
Comment (dereuromark) [inline tests/TestCase/Http/ServerRequestTest.php]: Those are still using the wrong argument order. "expected, result" is what it should be.

File: tests/TestCase/Http/ServerRequestTest.php [hunk 5]
```diff
@@ -3520,6 +3520,40 @@ public function testWithRequestTarget()
...
+     *
+     * @return void
+     */
+    public function testWithEnv()
+    {
+        $request = new ServerRequest();
+
+        $newRequest = $request->withEnv('HTTP_HOST', 'cakephp.org');
+        $this->assertNotSame($request, $newRequest);
+        $this->assertSame($newRequest->getEnv('HTTP_HOST'), 'cakephp.org');
```
Comment (dereuromark) [inline tests/TestCase/Http/ServerRequestTest.php]: wrong order.

**Pull request merged**